pkg/virt-controller/watch/vmi.go

```diff
@@ -573,13 +584,35 @@ func (c *VMIController) filterMatchingPods(vmi *virtv1.VirtualMachineInstance, p
 	return matchingPods, nil
 }
 
+func (c *VMIController) filterMatchingHandlerPods(launcherPod *k8sv1.Pod, handlerPods []*k8sv1.Pod) ([]*k8sv1.Pod, error) {
+	selector, err := v1.LabelSelectorAsSelector(&v1.LabelSelector{MatchLabels: map[string]string{virtv1.AppLabel: "virt-handler"}})
+	if err != nil {
+		return nil, err
+	}
+	matchingPods := []*k8sv1.Pod{}
+
+	for _, pod := range handlerPods {
+		if selector.Matches(labels.Set(pod.ObjectMeta.Labels)) && launcherPod.Status.HostIP == pod.Status.HostIP {
+			matchingPods = append(matchingPods, pod)
+		}
+	}
+	return matchingPods, nil
+}
+
 func isPodOwnedByHandler(pod *k8sv1.Pod) bool {
-	if pod.Annotations != nil && pod.Annotations[virtv1.OwnedByAnnotation] == "virt-handler" {
+	if pod.OwnerReferences != nil && pod.OwnerReferences[0].Kind == "Pod" && strings.Contains(pod.OwnerReferences[0].Name, "virt-handler") {
 		return true
 	}
 	return false
 }
 
+func updatePodOwnerToHandlerPod(pod, handlerPod *k8sv1.Pod) {
```

Comment:
Hm, that approach looks in this context wrong. `virt-handler` is then responsible, that's true but still the VMI object is the owner of the pod. I am not sure if we can replace the annotation.